src/lib/add-to-context-extractor/extractor.spec.ts

```diff
@@ -544,6 +545,119 @@ it('support async/await', () => {
   `)
 })
 
+describe('top-level union types', () => {
```

Comment:
How about a test with input like `{ a: 1 } | { b: 2 }` which I would expect to unify as: `{ a?: 1, b?: 2 }`.

Reply:
So, you might have caught a bug here. `{ a: 1 } | { b: 2 }` would be unified as `{ a: 1, b: 2 }` if "casted" like you just suggest. Because doing `return X as { a: 1 } | { b: 2 }` will tell TS to use this type as is and thus the `undefined`s won't be added

Reply:
Ah, agreed!

Maybe wrong, but looks low effort to handle that case?

Reply:
Well, I'm not sure what's the right approach to fix that edge-case. One way could be:

- if a property is not among all the union type members, then force it to be optional

🤔

Reply:
Yep sounds right to me

Reply:
Beautiful, seems to be working! Thanks for catching that 💯
![image](https://user-images.githubusercontent.com/25233379/86153355-d8824980-bb01-11ea-880f-3599a07d597c.png)